webgoat-integration-tests/src/test/java/org/owasp/webgoat/IntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+package org.owasp.webgoat;
+
+import io.restassured.RestAssured;
+import io.restassured.config.RestAssuredConfig;
+import io.restassured.config.SSLConfig;
+import lombok.Getter;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.owasp.webwolf.WebWolf;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class IntegrationTest {
+
+    private static String WEBGOAT_URL = "http://localhost:8080/WebGoat/";
+    private static String WEBWOLF_URL = "http://localhost:9090/";
+
+    //This also allows to test the application with HTTPS when outside testing option is used
+    protected static RestAssuredConfig restConfig = RestAssuredConfig.newConfig().sslConfig(new SSLConfig().relaxedHTTPSValidation());
+
+    @Getter
+    private String webGoatCookie;
+    @Getter
+    private String webWolfCookie;
+    @Getter
+    private String webgoatUser = UUID.randomUUID().toString();
+
+    private static boolean started = false;
+
+    @BeforeClass
+    public static void beforeAll() {
+        if (!started) {
+            started = true;
+            if (!areAlreadyRunning()) {
+                SpringApplicationBuilder wgs = new SpringApplicationBuilder(StartWebGoat.class)
+                        .properties(Map.of("spring.config.name", "application-webgoat"));
+                wgs.run();
+                SpringApplicationBuilder wws = new SpringApplicationBuilder(WebWolf.class)
+                        .properties(Map.of("spring.config.name", "application-webwolf"));
+                wws.run();
+            }
+        }
+    }
+
+    private static boolean areAlreadyRunning() {
+        return checkIfServerIsRunnningOn(9090) && checkIfServerIsRunnningOn(8080);
+    }
+
+    private static boolean checkIfServerIsRunnningOn(int port) {
```

Comment:
This check is incorrect. I have tested this on my environments. 
Something that would work is:
private static boolean checkIfServerIsRunnningOn(int port) {
        try (var ignored = new Socket("127.0.0.1", port)) {
            return true;
        } catch (IOException e) {
            return false;
        }
    }

Reply:
The code starts a `ServerSocket` on the port so if it fails the IOException would occur, meaning that there is something already running on that port.

Reply:
Sorry, but that check based on ServerSocket is really not working. It just returns falls when port 8080 is already taken and then starting WebGoat will still fail. You probably need to do something like a bind on a ServerSocket for it to really fail. The variant in my comment with Socket works as expected.

Reply:
Moved to using Socket as suggested above, the code works on Windows but seems to fail on Linux.